src/history/hs.c

```diff
@@ -666,6 +667,15 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
             if (upd->txnid == WT_TXN_ABORTED)
                 continue;
             WT_ERR(__wt_modify_vector_push(&modifies, upd));
+
+            /* Mark the second update after the prepared update. */
```

Comment:
In case if we agree to go with this approach, this definitely needs some long comment and reordering the conditions into specific brackets may make it easier for the code readability.